### PR TITLE
fix(bootstrap): guard SAM2 return trap cleanup

### PR DESCRIPTION
**Pull request by RC219805**
## Summary
- guard the SAM2 standard-install RETURN trap so a successful path does not fail on an unset `error_log`
- add a regression test that proves the Darwin arm64 SAM2 success path stays zero-exit and does not emit an `unbound variable` failure

## Validation
- `bash -n scripts/bootstrap/install_ml_stack.sh`
- `TMPDIR=/private/tmp /Users/richardcheetham/Desktop/Transformation_Portal/.venv/bin/pytest -q tests/validation/test_python_bootstrap_scripts.py -k return_trap --basetemp=/private/tmp/tp-pytest-return-trap`
- `PRE_COMMIT_HOME=/tmp/pre-commit PATH=/Users/richardcheetham/Desktop/Transformation_Portal/.venv/bin:$PATH TP_LINT_PYTHON=/Users/richardcheetham/Desktop/Transformation_Portal/.venv/bin/python /Users/richardcheetham/Desktop/Transformation_Portal/.venv/bin/pre-commit run --files scripts/bootstrap/install_ml_stack.sh tests/validation/test_python_bootstrap_scripts.py`

## Scope
- only `scripts/bootstrap/install_ml_stack.sh`
- only `tests/validation/test_python_bootstrap_scripts.py`

### CODE REVIEW

File: scripts/bootstrap/install_ml_stack.sh [hunk 1]
```diff
@@ -396,7 +396,7 @@ install_profile() {
                 # Create secure temporary file for error logging
                 local error_log
                 error_log="$(mktemp)"
-                trap 'rm -f "${error_log}"' RETURN
+                trap 'if [[ -n "${error_log:-}" ]]; then rm -f "${error_log}"; fi' RETURN
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/bootstrap/install_ml_stack.sh:402]: Addressed. The SAM2 branch now registers the cleanup trap with the resolved `mktemp` path, explicitly removes the error log on both the standard-success and fallback paths, and clears the `RETURN` trap immediately after cleanup so the function does not leak trap state into later profile installs. I also tightened the regression to stub `mktemp` and assert the temp file is actually gone after a successful `--profile sam2` run. Validated with `bash -n scripts/bootstrap/install_ml_stack.sh`, the targeted `pytest -k return_trap` regression, and targeted `pre-commit` on the touched files.

**Review by RC219805 — COMMENTED** [review]
review comments addressed, threads resolved.
